lxd/db/images.go

```diff
@@ -889,3 +889,67 @@ func (c *Cluster) ImageUploadedAt(id int, uploadedAt time.Time) error {
 	err := exec(c.db, "UPDATE images SET upload_date=? WHERE id=?", uploadedAt, id)
 	return err
 }
+
+// ImageGetNodesHasImage returns the list of the address of online nodes
+// which already have the image.
+// Note: the local address is not included in the returned address list.
+func (c *Cluster) ImageGetNodesHasImage(fingerprint string) ([]string, error) {
+	return c.getNodesByImageFingerprint(fingerprint, true)
+}
+
+// ImageGetNodesHasNoImage returns the list of the address of online nodes
+// which don't have the image.
+// Note: the local address is not included in the returned address list.
+func (c *Cluster) ImageGetNodesHasNoImage(fingerprint string) ([]string, error) {
+	return c.getNodesByImageFingerprint(fingerprint, false)
+}
+
+func (c *Cluster) getNodesByImageFingerprint(fingerprint string, hasImage bool) ([]string, error) {
+	q := `
+SELECT nodes.address FROM nodes
```

Comment:
I suspect that you need ```DISTINCT``` here, to avoid getting duplicate addresses in case ```hasImage``` is ```false```.